src/main/java/xyz/openmodloader/util/EnumUtils.java

```
@@ -0,0 +1,194 @@
+     */
+    private static <T extends Enum<?>> T makeEnum(Class<T> enumClass, String enumName, int ordinal, Class<?>[] additionalTypes, Object[] additionalValues) throws Exception {
+        Object[] params = new Object[additionalValues.length + 2];
+        params[0] = enumName;
+        params[1] = ordinal;
+        System.arraycopy(additionalValues, 0, params, 2, additionalValues.length);
+        return enumClass.cast(newInstance.invoke(getConstructorAccessor(enumClass, additionalTypes), new Object[] { params }));
+    }
+
+    /**
+     * Adds a new constant field to an Enum.
+     * 
+     * @param enumClass The class to add the new field to.
+     * @param enumName The name of the new field. Should be all upper cased and
+     *        unique.
+     * @param paramTypes An array of classes that represents the additional
+     *        parameters in the constructor.
+     * @param paramValues The values for every parameter in the enum
+     *        constructor.
+     * @return An object that represents the newly created enum field.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends Enum<?>> T addEnum(Class<T> enumClass, String enumName, final Class<?>[] paramTypes, Object[] paramValues) {
+
+        if (!Enum.class.isAssignableFrom(enumClass)) {
+            OpenModLoader.getLogger().warn("Class " + enumClass + " is not a valid Enum!");
+            return null;
+        }
+
+        Field valuesField = null;
+        Field[] fields = enumClass.getDeclaredFields();
+
+        for (Field field : fields) {
+            final String name = field.getName();
+            if (name.equals("$VALUES") || name.equals("ENUM$VALUES")) {
+                valuesField = field;
+                break;
+            }
+        }
+
+        if (valuesField == null) {
+            OpenModLoader.getLogger().warn("Could not find the $VALUES or ENUM$VALUES field in " + enumClass.getName() + ". Things will not work correctly!");

```

I would throw an exception here, for the same reason as above.